creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -377,37 +388,73 @@ class SnapbackSM {
+        if (clockValue === -1) {
+          newReplicaNodesSet.add(randomHealthyNode)
+        }
       } catch (e) {
         // Something went wrong in checking clock value. Reselect another secondary.
-        this.logError(e.message)
+        this.logError(`[determineNewReplicaSet] ${e.message}`)
       }
     }
 
-    if (currentNodeHasExistingUserState) throw new Error(`Unable to select new secondary given current healthy Content Nodes=[${healthyNodes}]`)
-
-    const currentHealthySecondary = unhealthyReplicasSet.has(secondary1) ? secondary2 : secondary1
+    let newReplicaNodes = Array.from(newReplicaNodesSet)
+    // We need a minimum of `NUMBER_OF_REPLICA_SET_NODES` to perform the reconfig op. In the case that
+    // the max number replica set attempts were made, push null
+    while (newReplicaNodes.length < NUMBER_OF_REPLICA_SET_NODES) {
+      newReplicaNodes.push(null)
+    }
 
-    // NOTE: For v0 implementation, `newSecondary2` will be the newly selected secondary, whilst the other nodes will be the
-    // original replica set primary and secondary1
-    return {
-      newPrimary: primary,
-      newSecondary1: currentHealthySecondary,
-      newSecondary2: newSecondary
+    // Note: for v0, only issue reconfigs if only 1 secondary is unhealthy. Else, just log out the tentative
+    // new replica set to be assigned.
+
+    if (unhealthyReplicasSet.size === 1) {
+      if (unhealthyReplicasSet.has(primary)) {
+        // If the primary is unhealthy, select the higher clock value of the two secondaries as the new primary,
+        // and then seelct 2 new secondaries
+        const secondariesToClockMap = await this.fetchClockValues(secondary1, secondary2, wallet)
+        newPrimary = secondariesToClockMap[secondary1] >= secondariesToClockMap[secondary2] ? secondary1 : secondary2
+        newSecondary1 = newReplicaNodes[0] // sync
+        newSecondary2 = newReplicaNodes[1] // sync
+      } else {
+        // If one secondary is unhealthy, select a new secondary
+        issueReconfig = true
+        const currentHealthySecondary = !unhealthyReplicasSet.has(secondary1) ? secondary1 : secondary2
+        newPrimary = primary
+        newSecondary1 = currentHealthySecondary
+        newSecondary2 = newReplicaNodes[0]
+      }
+    } else if (unhealthyReplicasSet.size === 2) {
+      if (unhealthyReplicasSet.has(primary)) {
+        // If primary + secondary is unhealthy, use other healthy secondary as primary and 2 random secondaries
+        newPrimary = !unhealthyReplicasSet.has(secondary1) ? secondary1 : secondary2
+      } else {
+        // If both secondaries are unhealthy, keep original primary and select two random secondaries
+        newPrimary = primary
+      }
+      newSecondary1 = newReplicaNodes[0] // sync
+      newSecondary2 = newReplicaNodes[1] // sync
+    } else {
+      // If entire replica set is unhealthy, select an entire new replica set
```

Comment:
this is super dangerous, i'm not sure snapback should be handling this case at all as it's a full data loss case. for now maybe just set it to the same replicaset and leave issueReconfig always false

Reply:
if we want to replace the entire replica set in the future, I think we should keep this chunk of code here. `issueReconfig` will be `false` in this flow, so no actual reconfig will occur. just the log statement

Reply:
or, I can set the new replica set values to `null` to be super sure that no reconfig will occur

Reply:
discussed offline - do nothing if entire rset is unhealthy